frontend/src/components/ShootDns/GDnsProviderRow.vue

```diff
@@ -176,58 +164,85 @@ export default {
       }),
     }
   },
-  data () {
-    return {
-      secretValid: true,
-    }
-  },
   computed: {
     ...mapState(useGardenerExtensionStore, [
       'dnsProviderTypes',
     ]),
-    dnsProvider () {
-      return this.dnsProviders[this.dnsProviderId]
-    },
-    primary () {
-      return this.dnsPrimaryProviderId === this.dnsProviderId
-    },
     dnsSecrets () {
       return this.dnsSecretsByProviderKind(this.dnsProviderType)
     },
-    typeHint () {
-      return this.primary && !this.isNewCluster
-        ? 'Primary Provider type cannot be changed after cluster creation'
-        : ''
-    },
-    primaryReadonly () {
-      return !this.isNewCluster && this.primary
-    },
     dnsProviderType: {
       get () {
-        return this.dnsProvider.type
+        return this.dnsProvider?.type
       },
       set (value) {
         this.dnsProvider.type = value
         const dnsSecrets = this.dnsSecretsByProviderKind(value)
         const defaultDnsSecret = head(dnsSecrets)
-        this.dnsProvider.secretName = get(defaultDnsSecret, 'metadata.secretRef.name', null)
+        this.extensionDnsProviderSecret = defaultDnsSecret
       },
     },
-    dnsProviderSecret: {
+    extensionDnsProviderSecret: {
```

Comment:
The name we used the the dns service provider is `dnsServiceExtensionProvider`. This property should have the name `dnsServiceExtensionProviderSecret` or shorter `dnsProviderSecret`

Reply:
I sill think this name is not in line with the convention we use for other names for `shoot-dns-service` provider properties?